src/Nancy/IO/RequestStream.cs

```diff
@@ -358,9 +358,15 @@ private void ShiftStreamToFileStreamIfNecessary()
 
         private static FileStream CreateTemporaryFileStream()
         {
-            var filePath = Path.GetTempFileName();
-
-            return new FileStream(filePath, FileMode.Create, FileAccess.ReadWrite, FileShare.None, 8192, StaticConfiguration.AllowFileStreamUploadAsync);
+			var filePath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString("N") + ".tmp");

```

Comment:
Can we just add a comment stating why we're doing this? Otherwise, you can bet someone comes along a year from now with a PR saying "why don't you just use `Path.GetTempFileName` and no one remembers :stuck_out_tongue: